http/logical.go

```diff
@@ -90,6 +91,10 @@ func buildLogicalRequest(core *vault.Core, w http.ResponseWriter, r *http.Reques
 
 func handleLogical(core *vault.Core, dataOnly bool, prepareRequestCallback PrepareRequestFunc) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Bubble up CORS preflight requests to the CORS handler.
```

Comment:
In my mind, a preflight won't even make it to the logical handler -- there's no need for it to. If all requests are passing through the wrapping handler, a very simple thing to do would be to just check for the operation being `OPTIONS` then and there and act appropriately.